(SOC-469) Only run this hook if Wall is enabled

Grunny opened the pull request:
Message Wall and Forum are intertwined right now so that if Forum is
enabled but Wall is disabled, then Message Wall's setup file and hooks
are still loaded, so we need this check in the hook method after all
as a quick fix...

/cc @jsutterfield @lizlux

Grunny committed:
(SOC-469) Only run this hook if Wall is enabled

Message Wall and Forum are intertwined right now so that if Forum is
enabled but Wall is disabled, then Message Wall's setup file and hooks
are still loaded, so we need this check in the hook method after all
as a quick fix.

armonr commented:
:shipit:

lizlux commented:
What a PITA! Thanks for fixing @Grunny

lizlux commented on extensions/wikia/Wall/WallHooksHelper.class.php:
```diff
@@ -2218,7 +2218,11 @@ public static function onArticleCommentGetSquidURLs( $title, &$urls ) {
 	 */
 	public static function onGetTalkPage( Title $title, Title &$talkPageTitle ) {
-		if ( !$title->isSubpage() && $title->getNamespace() == NS_USER ) {
+		global $wgEnableWallExt;
```
I generally opt for `F::app()->wg` rather than the global.

Grunny replied:
I've never had a strong preference for that if the code isn't in a Nirvana controller, since `F::app()` is still a global ;), but switched since it's consistent with everywhere else in Wall. :)

lizlux commented:
Also, might be worth adding a comment that these hooks are called for wall and forum, regardless of whether wall is enabled.

jsutterfield commented:
I agree with Liz that adding some docs as to why this logic is added would be good. Otherwise, LGTM. Thanks @Grunny!

Grunny committed:
Add comment of when Wall hooks will load

Add documentation explaining that Wall hooks will load if Forums are
enabled even if Wall is disabled.

Grunny commented:
Added a comment at the top, wasn't really sure where best to put it. :)
